### PR TITLE
Issue 18530 v2

### PR DESCRIPTION
**Pull request by erickgonzalez**
Apply feedback to the new endpoint that gets all the messages of the system.

Feedback:

- Change name to `api/v2/languages/{language}/keys`
- If the language requested exists in dotcms, get also the language keys and language variables
- Update tests

### CODE REVIEW

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 1]
```diff
@@ -207,17 +216,59 @@ public Response getAllMessages (
...
+            if(currentLocale.getLanguage().equalsIgnoreCase(locale.getLanguage())){
+                if(UtilMethods.isSet(locale.getCountry())){
+                    if(currentLocale.getCountry().equalsIgnoreCase(locale.getCountry())){
+                        break;
+                    }
+                }else{
+                    break;
+                }
+
+            } else if(i == (locales.length-1)){
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:239]: Codacy Issue found: [Useless parentheses.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5671019)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 2]
```diff
@@ -26,13 +30,18 @@
...
 import com.rainerhahnekamp.sneakythrow.Sneaky;
+import io.vavr.control.Try;
+import java.util.ArrayList;
+import java.util.HashMap;
 import java.util.List;
 import java.util.Locale;
 import java.util.Map;
+import java.util.TreeMap;
+import java.util.stream.Collectors;
+import java.util.stream.Stream;
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:41]: Codacy Issue found: [Avoid unused imports such as 'java.util.stream.Stream'](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5671019)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 3]
```diff
@@ -207,17 +216,59 @@ public Response getAllMessages (
...
+        if(UtilMethods.isSet(language1)) {
+            //Language Keys
+            final Map mapLanguageKeys = APILocator.getLanguageAPI()
+                    .getLanguageKeys(locale.getLanguage()).stream().collect(
+                            Collectors.toMap(LanguageKey::getKey, LanguageKey::getValue));
+
+            result.putAll(mapLanguageKeys);
+
+            //Language Variable
+            long langId = language1.getId();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:261]: Codacy Issue found: [Local variable 'langId' could be declared final](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5671019)

File: dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java [hunk 4]
```diff
@@ -207,17 +216,59 @@ public Response getAllMessages (
...
-        if(!UtilMethods.isSet(language1)){
-            final String message = "Language with Id or Code: "+ language + " does not exist";
-            Logger.error(this,message);
-            throw new IllegalArgumentException(message);
+        final InitDataObject initData = new WebResource.InitBuilder(webResource)
+                .requiredAnonAccess(AnonymousAccess.READ)
+                .requestAndResponse(request, response)
+                .rejectWhenNoUser(false).init();
+
+        final User user = initData.getUser();
```
Comment (dev-dotcms) [inline dotCMS/src/main/java/com/dotcms/rest/api/v2/languages/LanguagesResource.java:224]: Codacy Issue found: [Avoid declaring a variable if it is unreferenced before a possible exit point.](https://app.codacy.com/manual/dotCMS/core/pullRequest?prid=5671019)

**Review by freddyucv — APPROVED** [review]

**Review by nollymar — APPROVED** [review]

**Pull request merged**